Tasks/PowerShellV2/AccessTokenHelper.ps1

```diff
@@ -0,0 +1,205 @@
+Import-Module Microsoft.PowerShell.Security
+Import-Module $PSScriptRoot\ps_modules\VstsTaskSdk
+
+function Get-VstsFederatedTokenPS2Task {
+    param(
+        [Parameter(Mandatory=$true)]
+        $taskDict,
+        [Parameter(Mandatory=$true)]
+        [string]$vstsAccessToken
+    )
+
+    $serviceConnectionId = $taskDict["ConnectedServiceName"] 
+    $uri = $taskDict["Uri"] 
+    $planId = $taskDict["PlanId"]
+    $jobId = $taskDict["JobId"]
+    $hub = $taskDict["Hub"]
+    $projectId = $taskDict["ProjectId"]
+
+    $url = $uri + "$projectId/_apis/distributedtask/hubs/$hub/plans/$planId/jobs/$jobId/oidctoken?serviceConnectionId=$serviceConnectionId&api-version=7.1-preview.1"
+
+    $headers = @{
+        "Authorization" = "Basic " + [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":$($vstsAccessToken)"))
+        "Content-Type"  = "application/json"
+    }
+    
+    # POST request to generate the OIDC token
+    $response = Invoke-WebRequest -Uri $url -Method Post -Headers $headers -Body $body
+    $responseContent = $response.Content | ConvertFrom-Json
+    $oidcToken = $responseContent.oidcToken
+
+    if ($null -eq $oidcToken -or $oidcToken -eq [string]::Empty) {
+        throw (New-Object System.Exception("CouldNotGenerateOidcToken"))
```

Comment:
Looks like a very generic Exception type to me. Maybe we can introduce new Exception class for that purpose?